tests/eip712_messages/valid_py_and_ethers.json

```diff
@@ -0,0 +1,220 @@
+{
+    "valid_space_in_custom_type_metamask_returns_empty_string_if_non_alphanum_in_custom_type": {
```

Comment:
More of a question. Is this something we _want_ to allow? Is there a benefit I'm not seeing? It seems to me like spaces aren't necessarily _meant_ to be in these definitions. Does `ethers.js` go out of its way to make sure that this is valid?

Curious on the reasoning here as to why we'd want this in `py`.

Reply:
Source of inspiration for the test is here: https://ethereum.stackexchange.com/questions/153221/how-to-sign-message-eip-712-with-whitespace-in-name

EIP712 says a type should be a "valid identifier", which in solidity-speak is only _, &, and alphanumerics. 

Ethers pays no attention to this - if it's a valid string in JS, it encodes it:
https://github.com/ethers-io/ethers.js/blob/e454afb2fa3612c273a09b154c36b35fafab17b1/src.ts/hash/typed-data.ts#L172

Metamask uses a faulty regex in a possible attempt to match the "valid identifier". Faulty, because if it finds a non-alphanumeric, instead of throwing an error it just returns an empty string.
https://github.com/MetaMask/eth-sig-util/blob/10206bf2f16f0b47b1f2da9a9cfbb39c6a7a7800/src/sign-typed-data.ts#L434

Given that these messages are JSON-formatted, it seems to me most helpful to the user to allow any valid JSON string in an identifier.

Reply:
> Faulty, because if it finds a non-alphanumeric, instead of throwing an error it just returns (and then encodes) an empty string.

Yikes, yeah. Fair enough. Thanks for the details 👍🏼